### PR TITLE
Remove misc. references to undefined Kconfig symbols

### PR DESCRIPTION
**Pull request by ulfalizer**
I'm about to add [a CI check](https://github.com/zephyrproject-rtos/ci-tools/pull/21) for references to undefined Kconfig symbols. This gets rid of some actual issues from the symbol whitelist.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
+1 for docs

File: doc/guides/test/sanitycheck.rst [hunk 1]
```diff
@@ -341,7 +341,7 @@ extra_configs: <list of extra configurations>
             depends_on: adc
           test_resolution_6:
             extra_configs:
-              - CONFIG_ADC_QMSI_SAMPLE_WIDTH=6
+              - CONFIG_LOG_DEFAULT_LEVEL=3
```
Comment (nashif) [inline doc/guides/test/sanitycheck.rst]: how did an ADC related Kconfig turn into a LOG_LEVEL?

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/guides/test/sanitycheck.rst (hunk 1 above)
Comment (ulfalizer) [reply doc/guides/test/sanitycheck.rst]: I just picked that random symbol that exists. What's a better example I could use?

Comment (ulfalizer) [reply doc/guides/test/sanitycheck.rst]: @nashif *ping*

Note that this is in documentation.

**Review by tyrantlizard77 — APPROVED** [review]

File: doc/guides/test/sanitycheck.rst (hunk 1 above)
Comment (ulfalizer) [reply doc/guides/test/sanitycheck.rst]: @nashif 

Changed it to this. Is that fine?

```
common:
  tags: drivers adc
tests:
  test:
    depends_on: adc
  test_async:
    extra_configs:
      - CONFIG_ADC_ASYNC=y
```

**Review by nashif — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**